daisy/workflow/machinetype.go

```diff
@@ -17,6 +17,27 @@ package workflow
 import (
 	"fmt"
 	"regexp"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
 )
 
 var machineTypeURLRegex = regexp.MustCompile(fmt.Sprintf(`^(projects/(?P<project>%[1]s)/)?zones/(?P<zone>%[1]s)/machineTypes/(?P<machinetype>%[1]s)$`, rfc1035))
+
+var machineTypes struct {
+	valid []string
+	mu    sync.Mutex
+}
+
+func checkMachineType(client compute.Client, project, zone, machineType string) error {
+	machineTypes.mu.Lock()
+	defer machineTypes.mu.Unlock()
+	if strIn(machineType, machineTypes.valid) {
+		return nil
+	}
+	if _, err := client.GetMachineType(project, zone, machineType); err != nil {
+		return err
+	}
+	machineTypes.valid = append(machineTypes.valid, machineType)
```

Comment:
Should this be the full partial URL of the machine type? projects/p/zones/z/machineTypes/m?

The name alone is ambiguous and not all zones have the same machine types.